apps/omg_watcher/lib/omg_watcher/event_fetcher.ex

```diff
@@ -0,0 +1,349 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+defmodule OMG.Watcher.EventFetcher do
```

Comment:
I know that you're trying to decouple here, but the `EventFetcher` is a core service now and I think this should be done once (in `omg`) and cater for the needs of both the watcher and the child chain server.

It has a well defined behavior and scope of responsibilities (and a test) and it does the exact same service for both apps. The differences that set them apart (list of events I presume) should be abstracted out.

Reply:
so...  I was thinking what to do. the long term plan is to split out watcher and child chain. maybe the plan is to share "the engine" or we split out these things into dependencies. 
so basically, I didn't really know what the plan is going forward, so I decided to not couple it. but happy to put it in the same bucket... wdyt @achiurizo @unnawut ?

Reply:
ok so the feedback I got on slack is that we move forward with this approach for now and find a middle ground later when we decide how we're going to split watchers and child chains